autoload/go/path.vim

```diff
@@ -139,11 +139,6 @@ function! go#path#CheckBinPath(binpath)
     " remove whitespaces if user applied something like 'goimports   '
     let binpath = substitute(a:binpath, '^\s*\(.\{-}\)\s*$', '\1', '')
 
-    " if it's in PATH just return it
-    if executable(binpath) 
-        return binpath
-    endif
-
```

Comment:
I think it should use it from PATH. Can you please change it back? PATH is universal and it's ok to read it from there. It also makes the logic clear.